packages/server/src/websocket/websocket.service.ts

```diff
@@ -391,37 +392,59 @@ export class WebsocketService {
       if (response === 'OK') {
         this.logger.log('[addLiveSession] LiveSession added successfully', { connectionId })
 
-        // Subscribes to the Redis channel for the connection ID
+        // Use the general Redis connection (not the subscriber) to check subscribed channels
+        const subscribedChannels = await this.redis.pubsub('CHANNELS')
+        const isSubscribed = subscribedChannels.includes(connectionId)
+
+        this.logger.debug(`[addLiveSession] subscribedChannels: ${subscribedChannels} - isSubscribed: ${isSubscribed}`)
+
+        if (isSubscribed) {
+          // If already subscribed, unsubscribe first using the subscriber Redis connection
+          this.logger.log(`[addLiveSession] Already subscribed to ${connectionId}, unsubscribing first...`)
+          await this.redisSubscriber.unsubscribe(connectionId, (err, count) => {
+            if (err) this.logger.error(err.message)
+            this.logger.log(`Unsubscribed ${count} from ${connectionId} channel.`)
+          })
+          await new Promise((resolve) => setTimeout(resolve, 200)) // Optional delay
+        }
+
+        this.logger.debug('[addLiveSession] try subscribe')
+
+        // Use redisSubscriber for subscriptions only
         await this.redisSubscriber.subscribe(connectionId, (err, count) => {
           if (err) this.logger.error(err.message)
           this.logger.log(`Subscribed ${count} to ${connectionId} channel.`)
         })
 
-        // Handles messages received on the subscribed Redis channel
-        this.redisSubscriber.on('message', (channel: string, message: string) => {
-          if (channel === connectionId) {
-            this.logger.log(`*** [redisSubscriber] Received message from ${channel}: ${message} **`)
-
-            const jsonRpcResponse: JsonRpcResponseSubscriber = {
-              jsonrpc: '2.0',
-              method: 'messageReceive',
-              params: {
-                connectionId,
-                message: JSON.parse(message),
-                id: dto.id,
-              },
+        this.logger.log(`Listener event count: ${this.redisSubscriber.listenerCount('message')}`)
+
+        // Handle messages received on the subscribed Redis channel
+        if (this.redisSubscriber.listenerCount('message') <= 1) {
+          this.redisSubscriber.on('message', (channel: string, message: string) => {
+            if (channel === connectionId) {
+              this.logger.log(`*** [redisSubscriber] Received message from ${channel}: ${message} **`)
+
+              const jsonRpcResponse: JsonRpcResponseSubscriber = {
+                jsonrpc: '2.0',
+                method: 'messageReceive',
```

Comment:
This is for another PR, but I think this event should be called `messagesReceived`, since it receives an array of QueuedMessage. By the way, params' `message` field should be renamed to `messages` in such case.


Also note that the event is not completely documented in `docs/message-pickup-repository-server.md` (params are not specified).

Reply:
Noted. After this PR, we will refactor these comments